### ISSUE TITLE
Any way to combine this project with SciJava input classes?

### ISSUE THREAD

**Issue by ctrueden** [scijava/ui-behaviour#2]
At a glance, there seems to be some duplication (as well as things on top).

See:
- https://github.com/scijava/scijava-common/tree/scijava-common-2.50.0/src/main/java/org/scijava/input
- https://github.com/scijava/scijava-common/tree/scijava-common-2.50.0/src/main/java/org/scijava/display/event/input
- https://github.com/scijava/scijava-common/tree/scijava-common-2.50.0/src/main/java/org/scijava/display/event/window

**Comment by tpietzsch** [comment]
@ctrueden Would be nice. Let's talk about it at the hackathon?

**Comment by tpietzsch** [comment]
@ctrueden I had a very quick look. The scijava input stuff duplicates the AWT constructs (to add a layer of abstraction), right? Could you point me to where the translation from AWT events happens?

Actually I can see no duplication between scijava and configurable-keys. The configurable-keys project is about `Behaviours` which do not care by which keys / mouse buttons / modifiers they are triggered. The idea is to make that configurable. There is the `MouseAndKeyHandler` which is all kinds of AWT listeners and maps AWT events to behaviours. We could certainly have a similar handler that listens to scijava events and maps those to behaviours. I was also thinking of making such a listener for jogl's NEWT, such that it could be potentially used in ClearVolume.

**Comment by ctrueden** [comment]
> Could you point me to where the translation from AWT events happens?

It is the [scijava-ui-awt component](https://github.com/scijava/scijava-ui-awt/tree/scijava-ui-awt-0.1.4/src/main/java/org/scijava/ui/awt).

> We could certainly have a similar handler that listens to scijava events and maps those to behaviours.

Right, I think the SciJava API could benefit from behaviors. In any case, this idea seems general enough to go into the scijava org instead of just bigdataviewer, don't you think?

> Let's talk about it at the hackathon?

Yeah, great!

**Comment by tpietzsch** [comment]
@ctrueden Yes, we were also thinking it should be in either scijava or imglib2-ui. At the moment, this is pre-alpha stuff, for development of TrackMate3. Putting it here was the no-bureaucracy option. We can move it later

**Comment by ctrueden** [comment]
We can move it into SciJava any time if that's what you and @tinevez want. Just say the word.
